app/models/flood_risk_engine/address.rb

```diff
@@ -33,5 +35,18 @@ def parts
     def address_methods
       %i[premises street_address locality city postcode]
     end
+
+    private
+
+    def clean_up_duplicate_addresses
+      return unless addressable.is_a?(FloodRiskEngine::Organisation)
+
+      primary_addresses = FloodRiskEngine::Address.where(addressable: addressable)
```

Comment:
Will this delete all the addresses for this organisation? 'Primary' suggests there might be 'secondary', etc. - or is that not relevant?

Reply:
There is code for 9(!) types of address:

```
      primary: 0,
      secondary: 1,
      site: 2,
      contact: 3,
      billing: 4,
      principal_office: 5,
      principal_business: 6,
      registered_office: 7,
      partner: 8
```

But from looking at the data, the only one we actually ever seem to use is the primary one – I think a lot of this was in there in case the db was ever hooked up with the CRM. We definitely don't have a need for a billing address in a free service, for example. The only other one that there's evidence of use is the contact address, although there doesn't appear to be a way to add that any more, and the addressable class for that is the Contact. So I think we're OK but I'll add the `address_type` check in as well just to be safe.

Reply:
I think there was a hefty cut-and-paste when this app was built!